modules/mvt/src/lib/parse-mvt.js

```diff
@@ -40,9 +48,17 @@ function getDecodedFeature(feature, options = {}) {
     throw new Error('MVT Loader: WGS84 coordinates need tileIndex property. Check documentation.');
   }
 
-  if (wgs84Coordinates && hasTileIndex) {
-    return feature.toGeoJSON(options.tileIndex.x, options.tileIndex.y, options.tileIndex.z);
+  const decodedFeature =
+    wgs84Coordinates && hasTileIndex
+      ? feature.toGeoJSON(options.tileIndex.x, options.tileIndex.y, options.tileIndex.z)
+      : transformCoordinates(feature, transformToLocalCoordinates);
+
+  // Add layer name to GeoJSON properties
+  const hasLayerProperty = decodedFeature.properties.hasOwnProperty('layer');
+  const layerPropertyName = hasLayerProperty ? options.layerProperty : 'layer';
```

Comment:
Concern: My thought is that if `options.mvt.layerProperty` is provided, we always use that. This code seems to be using the user specified option name only if the `layer` prop is already defined on the tile. It will be hard for the application to know what prop to looks at.